lib/osml/data_intake/di_dataplane.ts

```diff
@@ -0,0 +1,210 @@
+   */
+  securityGroupId?: string;
+
+  /**
+   * The IAM (Identity and Access Management) role to be used for Lambda (optional).
+   * @type {IRole | undefined}
+   */
+  lambdaRole?: IRole;
+
+  /**
+   * Custom configuration for the DIDataplane Construct (optional).
+   * @type {DIDataplaneConfig | undefined}
+   */
+  config?: DIDataplaneConfig;
+}
+
+/**
+ * Represents the DIDataplane construct responsible for managing the data plane
+ * of the model runner application. It handles various AWS resources and configurations
+ * required for the application's operation.
+ *
+ * @param {Construct} scope - The scope/stack in which to define this construct.
+ * @param {string} id - The id of this construct within the current scope.
+ * @param {DIDataplaneProps} props - The properties of this construct.
+ * @returns {DIDataplane} - The DIDataplane construct.
+ */
+export class DIDataplane extends Construct {
+  // Public properties
+  public config: DIDataplaneConfig;
+  public lambdaFunction: DockerImageFunction;
+  public inputTopic: Topic;
+  public outputTopic: Topic;
+  public outputBucket: Bucket;
+  public inputBucket: Bucket;
+  public removalPolicy: RemovalPolicy;
+  public securityGroup?: ISecurityGroup;
+  public lambdaRole?: IRole;
+  /**
+   * Constructs an instance of DIDataplane.
+   *
+   * @constructor
+   * @param {Construct} scope - The scope/stack in which to define this construct.
+   * @param {string} id - The id of this construct within the current scope.
+   * @param {DIDataplaneProps} props - The properties of this construct.
+   */
+  constructor(scope: Construct, id: string, props: DIDataplaneProps) {
+    super(scope, id);
+    // Setup class from base properties
+    this.setup(props);
+
+    // Setup a removal policy
+    this.removalPolicy = props.account.prodLike
+      ? RemovalPolicy.RETAIN
+      : RemovalPolicy.DESTROY;
+
+    // Create a test input bucket
+    this.inputBucket = new OSMLBucket(this, `DIOutputBucket`, {
+      bucketName: props.account.id + "-" + this.config.S3_INPUT_BUCKET_NAME,
+      prodLike: props.account.prodLike,
+      removalPolicy: this.removalPolicy
+    }).bucket;
+
+    // Create a test output bucket
+    this.outputBucket = new OSMLBucket(this, `DIInputBucket`, {
+      bucketName: props.account.id + "-" + this.config.S3_OUTPUT_BUCKET_NAME,
+      prodLike: props.account.prodLike,
+      removalPolicy: this.removalPolicy
+    }).bucket;
+
+    // Create an SNS topic
+    this.inputTopic = new Topic(this, "DIInputTopic", {
+      topicName: this.config.SNS_INPUT_TOPIC_NAME
+    });
+
+    // Create an SNS topic
+    this.outputTopic = new Topic(this, "DIOutputTopic", {
+      topicName: this.config.SNS_OUTPUT_TOPIC_NAME
+    });
+
+    // Define a Lambda function with a container image
+    this.lambdaFunction = new DockerImageFunction(
+      this,
+      "ImageHandlerFunction",
+      {
+        code: props.dockerImageCode,
+        timeout: Duration.seconds(120),
```

Comment:
Consider making this timeout a parameter. Also test it with some of the Maxar NITFs. The time to create overviews and generate image statistics for a big image might push up against this timeout.

Reply:
done